Add Plugin Submission Documentation (Rewrite of the old)

MiguVT opened the pull request:
As i told, i made a fast doc about submission yesterday, i took some time on my break time and made this rewrite, and not used AI for fix things, hope it helps :b

MiguVT committed:
Add Plugin Submission Guide to Official Documentation

- Created a detailed guide for submitting plugins to Vencord.
- Outlined steps for forking the repository, adding a plugin, and creating a pull request.
- Included guidelines for adhering to Vencord's code structure and standards.
- Explained the review process and licensing requirements.

MiguVT committed:
Rewrite plugin submission guide for clarity and accuracy

This commit rewrites the plugin submission guide to remove unnecessary installation instructions, clarify that plugins are first merged into the dev branch, and ensure the correct submission process. Additionally, the guide now correctly links to the Vencord community Discord and removes redundant file listings, focusing on providing clear instructions for submitting pre-made plugins.

MiguVT committed:
Manually rewrited everything

I used my break time on my job to rewrite everything, hope it helps :b

MiguVT commented:
Aditionally, i added an update section, i can delete it if wanted

Sqaaakoi commented on src/content/docs/plugins/submission.md:
```diff
@@ -1,4 +1,115 @@
+
+1. **Fork the Repository**: Go to the specific repository on [https://github.com/Vendicated/Vencord](https://github.com/Vendicated/Vencord) page and select “Fork”. 
+
+2. **Clone Your Fork** 
```
This is too oversimplified; There are a decent amount of people who have decided to install userplugins before getting started. Change this to branch into a "if you've installed userplugins set the remote and upstream branches" and an "if not clone the fork".

MiguVT replied:
Fixed :D

Sqaaakoi commented on src/content/docs/plugins/submission.md:
```diff
@@ -1,4 +1,115 @@
+- The folder in which stores your plugins should have camelCase name (for instance, `myAwesomePlugin`). 
+- `README.md` is in your plugin folder with a description, usage instructions, screenshots(if any) 
+- The coding of your plugin adheres to Vencord’s coding standards 
+- This is the best because probably, you have engaged in rigorous testing of the plugin. 
```
...what?

MiguVT replied:
Fixed, i dont know what i wrote on there, i was a little sleepy 😅

Sqaaakoi commented on src/content/docs/plugins/submission.md:
```diff
@@ -1,4 +1,115 @@
+### Pre-submission Checklist 
+
+- The folder in which stores your plugins should have camelCase name (for instance, `myAwesomePlugin`). 
+- `README.md` is in your plugin folder with a description, usage instructions, screenshots(if any) 
```
Suggest users to attach images in GitHub issue comments. There's an issue thread for it

MiguVT replied:
Mentioned it 👍🏻

Sqaaakoi commented on src/content/docs/plugins/submission.md:
```diff
@@ -1,4 +1,115 @@
+### Common Pitfalls 
+
+- Please make sure that the folder name of the plugin is the same as the one used in the commit message and the PR title 
+- Ensure that you have only committed files only found within your plugin’s directory. 
```
There should probably be an "unless absolutely necessary" clause here

MiguVT replied:
Added :b

Sqaaakoi commented on src/content/docs/plugins/submission.md:
````diff
@@ -1,4 +1,115 @@
+
+5. **Push Your Changes** 
+```bash 
+git push -u origin update-plugin/yourPluginName-description 
````
Overly verbose? Maybe instruct users to learn git branching earlier

MiguVT replied:
Added a note to the end 😀

Sqaaakoi reviewed (commented):
It's very verbose regarding telling users just do [Git commands] rather than teaching the user how to use Git. Do note that this is the reason why support for userplugins is not given in support; users will follow the instructions without learning anything about the software they are using

MiguVT committed:
docs(Plugins): Update submission and update guide

- Improve clarity and organization of the plugin submission process
- Add separate instructions for users with existing Vencord installations
- Clarify common pitfalls and best practices
- Include guidance on attaching images to GitHub issues
- Add note about git branching knowledge

MiguVT commented:
Everything fixed, tysm for your comments and notes @Sqaaakoi @Sqaaakoi :b

cuboid101 commented:
This PR should either be merged or closed IMO.
The docs just have an empty page, something was seemingly meant to go there.

MiguVT commented:
> This PR should either be merged or closed IMO. The docs just have an empty page, something was seemingly meant to go there.

I agree, if no one accept this PR, i will close it.

MiguVT commented:
💖
